src/material-experimental/mdc-tabs/_tabs-common.scss

```diff
@@ -0,0 +1,148 @@
+@import '@material/ripple/variables';
+@import '../../material/core/style/variables';
+@import '../../material/core/style/noop-animation';
+@import '../../material/core/style/vendor-prefixes';
+@import '../../cdk/a11y/a11y';
+
+$mat-tab-animation-duration: 500ms !default;
+
+@mixin tab {
+  &.mdc-tab {
+    // MDC's tabs stretch to fit the header by default, whereas stretching on our current ones
+    // is an opt-in behavior. Also technically we don't need to combine the two classes, but
+    // we need the extra specificity to avoid issues with CSS insertion order.
+    flex-grow: 0;
+    min-width: 160px;
+  }
+
+  &::before {
+    content: '';
+    display: block;
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    opacity: 0;
+    pointer-events: none;
+  }
+
+  // We need to handle the hover and focus indication ourselves, because we don't use MDC's ripple.
+  &:hover::before {
+    opacity: map-get($mdc-ripple-dark-ink-opacities, hover);
+  }
+
+  &.cdk-program-focused,
+  &.cdk-keyboard-focused {
+    &::before {
+      opacity: map-get($mdc-ripple-dark-ink-opacities, focus);
+    }
+  }
+
+  .mat-ripple-element {
+    opacity: map-get($mdc-ripple-dark-ink-opacities, press);
+  }
+}
+
+@mixin tab-ripple {
+  .mat-mdc-tab-ripple {
+    position: absolute;
+    top: 0;
+    left: 0;
+    bottom: 0;
+    right: 0;
+    pointer-events: none;
+  }
+}
+
+// Structural styles for a tab header. Used by both `mat-tab-header` and `mat-tab-nav-bar`.
+@mixin paginated-tab-header {
+  .mat-mdc-tab-header {
```

Comment:
Add comments that explain why the styles defined here are necessary on top of the MDC styles? I want to make sure that each additional bit of CSS we add on top of MDC has an explanation of _why_ we added it. 

(in this case I assume it's b/c they don't support tab pagination)